src/HealthChecks.AzureApplicationInsights/AzureApplicationInsightsHealthCheck.cs

```diff
@@ -0,0 +1,74 @@
+using Microsoft.Extensions.DependencyInjection;
+
+namespace HealthChecks.AzureApplicationInsights
+{
+    public class AzureApplicationInsightsHealthCheck : IHealthCheck
+    {
+        //from https://docs.microsoft.com/en-us/azure/azure-monitor/app/ip-addresses#outgoing-ports
+        private readonly string[] _appInsightsUrls =
+        {
+            "https://dc.applicationinsights.azure.com",
+            "https://dc.applicationinsights.microsoft.com",
+            "https://dc.services.visualstudio.com"
+        };
+        private readonly string _instrumentationKey;
+
+        private readonly IHttpClientFactory _httpClientFactory;
+
+        public AzureApplicationInsightsHealthCheck(string instrumentationKey, IHttpClientFactory httpClientFactory)
+        {
+            _instrumentationKey = instrumentationKey ?? throw new ArgumentNullException(nameof(instrumentationKey));       
+            _httpClientFactory = httpClientFactory ?? throw new ArgumentNullException(nameof(httpClientFactory)); 
+        }
+
+        public async Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context, CancellationToken cancellationToken = default)
+        {
+            try
+            {
+                bool resourceExists = await ApplicationInsightsResourceExistsAsync(cancellationToken);
+                if (resourceExists)
+                {
+                    return HealthCheckResult.Healthy();
+                }
+                else
+                {
+                    return new HealthCheckResult(context.Registration.FailureStatus, description: $"Could not find application insights resource. Searched resources: {string.Join(", ", _appInsightsUrls)}");
+                }
+            }
+            catch (Exception ex)
+            {
+                return new HealthCheckResult(context.Registration.FailureStatus, exception: ex);
+            }
+        }
+
+        public async Task<bool> ApplicationInsightsResourceExistsAsync(CancellationToken cancellationToken)
+        {
+            using (var httpClient = _httpClientFactory.CreateClient(AzureApplicationInsightsHealthCheckBuilderExtensions.AZUREAPPLICATIONINSIGHTS_NAME))
+            {
+                string path = $"/api/profiles/{_instrumentationKey}/appId";
+                int index = 0;
+
+                while (index < _appInsightsUrls.Length)
+                {
+                    try
+                    {
+                        var uri = new Uri(_appInsightsUrls[index++] + path);
+                        HttpResponseMessage response = await httpClient.GetAsync(uri, HttpCompletionOption.ResponseHeadersRead, cancellationToken);
+                        if (response.IsSuccessStatusCode)
+                        {
+                            return true;
+                        }
+                    }
+                    catch (Exception e)
+                    {
+                        ExceptionDispatchInfo.Capture(e).Throw();
```

Comment:
I'm surprised a bit here. This call is pointless now, you may just type `throw;` or just remove try/catch block. What is more important - you began to throw the exception without iterating through all the values in `_appInsightsUrls`. As I understand it, the whole point of this healthcheck is to check the availability of **at least one** url. If the first is not available with some exception happens, but the rest are available, then code rethrows exception so the behavior is wrong.

Reply:
You're right regarding the iteration of all the values in `_appInsightUrls`. I added a block to only throw if we reached the end of the iteration, otherwise the loop will continue. Regarding your comment regarding that I could just `throw`, you said in an earlier [comment](https://github.com/Xabaril/AspNetCore.Diagnostics.HealthChecks/pull/740#discussion_r850306706) that it was necessary to preserve the stacktrace of the original exception. You want me to change that?

Reply:
It really **was** necessary when you stored exception in variable and then **after** some loop iterations rethrowed it. Now you catch exception and rethrow it immediately so all `ExceptionDispatchInfo.Capture(e).Throw();` call is pointless.